Cesium3DTilesSelection/src/CesiumIonTilesetLoader.cpp

```diff
@@ -12,6 +12,106 @@
 #include <unordered_map>
 
 namespace Cesium3DTilesSelection {
+
+// This is an IAssetAccessor decorator that handles token refresh for any asset
+// that returns a 401 error.
+class CesiumIonAssetAccessor
+    : public std::enable_shared_from_this<CesiumIonAssetAccessor>,
```

Comment:
Basically what happens is:
1. The shared_ptr captured by the lambdas guarantees the `CesiumIonAssetAccessor` lives at least as long as all of its in-flight requests.
2. The `_pTilesetLoader` being set to nullptr when the tileset loader is destroyed means it's fine for the `CesiumIonAssetAccessor` - and its requests - to outlive the `CesiumIonTilesetLoader`. Once the loader is destroyed (Tilest destruction, basically), tokens simply won't be refreshed anymore.

All of this might be a little unnecessarily elaborate. There are already mechanisms in place to make sure the TilesetContentManager, and therefore the TilesetContentLoader, can't be destroyed while requests are in flight. But it's also relatively unintrusive and adds a measure of safety. Plus it won't fail in the (unlikely?) event that requests not directly related to tile loads are in flight from this accessor.

Reply:
Ok, I didn't realize that `_pTilesetLoader` was set to `nullptr` in the destructor of `CesiumIonTilesetLoader`.